geps/gep-2162.md

```diff
@@ -0,0 +1,106 @@
+# GEP-2162: Supported features in GatewayClass Status
+
+* Issue: [#2162](https://github.com/kubernetes-sigs/gateway-api/issues/2162)
+* Status: Provisional
+
+## TLDR
+
+This GEP proposes to enhance the [GatewayClassStatus](https://github.com/kubernetes-sigs/gateway-api/blob/f2cd9bb92b4ff392416c40d6148ff7f76b30e649/apis/v1beta1/gatewayclass_types.go#L185) to include a list of Gateway API features supported by the installed GatewayClass. 
+
+## Goals
+
+* Improve UX by enabling users to easily see what features the implementation(GatewayClass) support.
+
+* Automatically run conformance tests based on the supported features populated in GatewayClass status
+
+* Provide foundation for tools to block or warn when unsupported features are used.
+
+## Non-Goals
+
+* Validate correctness of supported features published by the implementation. Meaning we don't intend to verify whether the supported features reported by the implementation are indeed supported.
+
+## Introduction
+
+The current [GatewayClassStatus](https://github.com/kubernetes-sigs/gateway-api/blob/f2cd9bb92b4ff392416c40d6148ff7f76b30e649/apis/v1beta1/gatewayclass_types.go#L185 is only used to store conditions the controller publishes.
+
+Partnered with the [Conformance Profiles](https://github.com/kubernetes-sigs/gateway-api/blob/main/geps/gep-1709.md) work, we want to:
+
+1. Improve UX by enabling users to easily see what features the implementation(GatewayClass) support.
+1. Automatically run conformance tests based on the supported features populated in GatewayClass status.
+1. Potentially build tooling to block or warn when unsupported features are used (more under [Future Work](#future-work)).
+
+This doc proposes to enhance the GatewayClassStatus API so implementations could publish a list of features they support/don't support.
+
+Implementations **must** publish the supported features before Accepting the GatewayClass.
+
+Implementations are free to decide how they manage this information - common approaches could be either to maintain static lists of supported features OR generate them from their conformance tests results/reports as part of their CI.
+
+## API
+
+This GEP proposes API changes describes as follow:
+1. Update the `GatewayClassStatus` struct to include a string-represented list of `SupportedFeatures`.
```

Comment:
Nit: Our markdown parser requires an empty newline before the start of a list. On that note, do we need a list here? It seems like there's only one item?
```suggestion
This GEP proposes API changes describes as follows:

1. Update the `GatewayClassStatus` struct to include a string-represented list of `SupportedFeatures`.
```

Reply:
changed it to "*"